src/main.py

```diff
@@ -38,69 +39,147 @@ def get_current_metric_definitions():
 
+
 def get_previous_metric_definitions():
     sys.stderr.write(f"Downloading metrics from s3://${METRICS_BUCKET}/{METRICS_KEY}... ")
     response = s3_client.get_object(Bucket = METRICS_BUCKET, Key = METRICS_KEY)
     print(f"Done", file=sys.stderr)
 
     return json.load(response['Body'])
 
+
 def save_metric_definitions(metric_names):
     body = json.dumps(metric_names)
 
     sys.stderr.write(f"Uploading metrics to s3://${METRICS_BUCKET}/${METRICS_KEY}... ")
-    s3_client.put_object(Body=body, Bucket = METRICS_BUCKET, Key = METRICS_KEY)
+    s3_client.put_object(Body = body, Bucket = METRICS_BUCKET, Key = METRICS_KEY)
     print(f"Done", file=sys.stderr)
 
-def get_percentage_change(url, metric_name, aggregationFunction):
-    df = pd.read_csv(url)
-
-    df['date'] = pd.to_datetime(df['date'])
-
-    area_names = df.areaName.unique()
-    ret = []
-
-    for area_name in area_names:
-        area_data = df[df.areaName.eq(area_name)]
-        
-        latest = df['date'].max()
-
-        # We want 7 days inclusive of the latest
-        six_days_before = latest - pd.Timedelta(days=6)
-        seven_days_before = latest - pd.Timedelta(days=7)
-        # As above
-        thirteen_days_before = latest - pd.Timedelta(days=13)
 
-        area_data_last_week = area_data[area_data.date.ge(six_days_before)]
-        area_data_week_before_last = area_data[area_data.date.ge(thirteen_days_before) & area_data.date.le(seven_days_before)]
+def get_populations():
+    url = "https://www.ons.gov.uk/file?uri=/peoplepopulationandcommunity/populationandmigration/populationestimates" \
+          "/datasets/populationestimatesforukenglandandwalesscotlandandnorthernireland" \
+          "/mid2019april2019localauthoritydistrictcodes/ukmidyearestimates20192019ladcodes.xls"
+    resp = requests.get(url)
+    df = pd.read_excel(io=resp.content, sheet_name="MYE2 - Persons", header=4, usecols="A,B,D", index_col=0)
 
-        aggregation_output_last_week = getattr(area_data_last_week[metric_name], aggregationFunction)()
-        aggregation_output_week_before = getattr(area_data_week_before_last[metric_name], aggregationFunction)()
-
-        percentage_change = ((aggregation_output_last_week - aggregation_output_week_before) / aggregation_output_week_before) * 100.0
-
-        ret.append([area_name, aggregation_output_week_before, aggregation_output_last_week, percentage_change])
+    return df
 
-    column_names = [
-        'areaName',
-        f'{metric_name}-{thirteen_days_before.strftime("%m-%d-%Y")}-to-{seven_days_before.strftime("%m-%d-%Y")}',
-        f'{metric_name}-{six_days_before.strftime("%m-%d-%Y")}-to-{latest.strftime("%m-%d-%Y")}',
-        'percentageChange'
-    ]
 
-    ret_df = pd.DataFrame(ret, columns = column_names)
-    return ret_df
+def population_for_area(populations_df, area_name, area_code):
+    try:
+        area_population = populations_df.at[area_code, "All ages"]
+        return area_population
+    except KeyError:
```

Comment:
This is the lookup error we only log at the moment

Reply:
what's the behaviour in this case? do we omit the row or just have a blank for a particular cell?

Reply:
looks like the function may return undefined (or Python equivalent) and then I'm not sure what line 79 would do

Reply:
Hmm yes you get a `RuntimeWarning` when it tries to divide by whatever it is (a None object?) and the whole thing blows up. I'll add some exception handling so that we return a result with the data we can fetch and a warning

Reply:
We now [handle this exception](https://github.com/guardian/uk-coronavirus-data-alerts/pull/4/commits/be16228927f758abee5dd6ef992468941a787067) to log that we've seen in and continue to process the rest of the results.